grapher/core/LegacyToOwidTable.ts

```diff
@@ -232,8 +360,189 @@ export const legacyToOwidTableAndDimensions = (
     return { dimensions: newDimensions, table: joinedVariablesTable }
 }
 
-const fullJoinTables = (tables: OwidTable[]): OwidTable =>
-    tables.reduce((joinedTable, table) => joinedTable.fullJoin(table))
+const fullJoinTables = (
+    tables: OwidTable[],
+    indexColumnNames: OwidTableSlugs[],
+    mergeFallbackLookupColumns?: OwidTableSlugs[][]
+): OwidTable => {
+    // This function merges a number of OwidTables together using a given list of columns
+    // to be used as the merge key. The merge key columns are used to construct a full set
+    // of index values from the various tables - all tables are enumerated, we create
+    // a merged string value from the index column values for each row and then we create
+    // a set from all these string values.
+    // Note that not every table has to have values for all columns - not even all the index
+    // columns have to exist on all tables (the index columns have to exist on the first table though)!
+    // The reason for this and how this can possibly work
+    // is that we also have a list of fallback merge columns. This is required so we can handle
+    // not just the easy case where we have year+entity for every table to be merged (which in our
+    // data model is by far the most common default), but also handle cases where we merge year and
+    // day based variables together. For this latter case we need to still contstruct the set of index
+    // values for the final table, but then when we try to look up values in the various tables to
+    // merge together we will not find values by day+entity for the year based tables. So for this
+    // case we get a series of fallback column tuples that we try in turn if the main index lookup
+    // fails. These fallback tuples are year+entity first and then entity only. The reasoning here is
+    // that e.g. when merging population to a day variable we want to merge the values from the year
+    // matching the day from the population variable (year+entity lookup) but for variables that don't
+    // have overlapping years (e.g. continents that only has 2015 as the single year) we want to fall back
+    // to merging by entity alone as a last resort
+    if (tables.length === 0) return new OwidTable()
+    else if (tables.length === 1) return tables[0]
+
+    // Get all the index values per table and then figure out the full set of all stringified index values
+    const indexValuesPerTable = tables.map((table) =>
+        // When we get a mergeFallbackLookupColumn then it can happen that a table does not have all the
+        // columns of the main index. In this case, just return an empty map because that will lead all
+        // lookups by main index to fail and we'll try the fallback index
+        !mergeFallbackLookupColumns ||
+        difference(indexColumnNames, table.columnSlugs).length === 0
```

Comment:
I had trouble understanding this condition, and negating it to `mergeFallbackLookupColumns && difference(indexColumnNames, table.columnSlugs).length > 0` in my head helped me a lot to better understand it. I campaign for turning the ternary around :)